utilities/pipelines/tokensReplacement/Convert-TokensInParameterFile.ps1

```diff
@@ -43,64 +37,52 @@ function Convert-TokensInParameterFile {
+        [hashtable] $Tokens = @{},
 
         [parameter(Mandatory = $true)]
-        [string]$TokenPrefix,
+        [string] $TokenPrefix,
 
         [parameter(Mandatory = $true)]
-        [string]$TokenSuffix,
+        [string] $TokenSuffix,
 
         [parameter(Mandatory = $false)]
-        [psobject]$OtherCustomParameterFileTokens,
+        [bool] $SwapValueWithName = $false,
 
         [parameter(Mandatory = $false)]
-        [bool]$SwapValueWithName = $false,
-
-        [parameter(Mandatory = $false)]
-        [string]$OutputDirectory
+        [string] $OutputDirectory
     )
 
     begin {
         # Load used funtions
         . (Join-Path $PSScriptRoot './helper/Convert-TokenInFile.ps1')
-        $AllCustomParameterFileTokens = @()
     }
 
     process {
-        Write-Verbose "Default Tokens Count: ($($DefaultParameterFileTokens.Count)) Tokens (From Input Parameter)"
-        ## Get Local Custom Parameter File Tokens (Should not Contain Sensitive Information)
-        Write-Verbose "Local Custom Tokens Count: ($($LocalCustomParameterFileTokens.Count)) Tokens (From Settings File)"
-        $AllCustomParameterFileTokens += ($LocalCustomParameterFileTokens | Select-Object -Property Name, Value)
-        # Combine All Input Token Types, Remove Duplicates and Only Select Name, Value if they contain other unrequired properties
-        $AllCustomParameterFileTokens = $DefaultParameterFileTokens + $LocalCustomParameterFileTokens |
-            ForEach-Object { [PSCustomObject]$PSItem } |
-            Sort-Object Name -Unique |
-            Select-Object -Property Name, Value |
-            Where-Object { $null -ne $PSitem.Name -and $null -ne $PSitem.Value }
-
-        # Other Custom Parameter File Tokens (Can be used for testing)
-        if ($OtherCustomParameterFileTokens) {
-            $AllCustomParameterFileTokens += $OtherCustomParameterFileTokens | ForEach-Object { [PSCustomObject]$PSItem }
+        # Combine All Input Token Types, Remove Duplicates and Only Select entries with on empty values
+        $FilteredParameterFileTokens = ($Tokens | Sort-Object -Unique).Clone()
+        @($FilteredParameterFileTokens.Keys) | ForEach-Object {
+            if ([String]::IsNullOrEmpty($FilteredParameterFileTokens[$_])) {
+                $FilteredParameterFileTokens.Remove($_)
+            }
         }
-        Write-Verbose ("All Parameter File Tokens Count: ($($AllCustomParameterFileTokens.Count))")
+        Write-Verbose ('Using [{0}] tokens' -f $FilteredParameterFileTokens.Keys.Count)
+
         # Apply Prefix and Suffix to Tokens and Prepare Object for Conversion
         Write-Verbose ("Applying Token Prefix '$TokenPrefix' and Token Suffix '$TokenSuffix'")
-        foreach ($ParameterFileToken in $AllCustomParameterFileTokens) {
-            $ParameterFileToken.Name = -join ($TokenPrefix, $ParameterFileToken.Name, $TokenSuffix)
+        foreach ($ParameterFileTokenName in @($FilteredParameterFileTokens.Keys)) {
```

Comment:
maybe remove the `ParameterFileToken` and just keep `TokenName` as variable name. It makes it more consistent with the orginal input parameter `Tokens`

Reply:
done